File: server-scala-play-server/app/model/SubscriptionOption.scala
package model

import play.api.libs.json._

/**
  * Represents the Swagger definition for SubscriptionOption.
  */
@javax.annotation.Generated(value = Array("org.openapitools.codegen.languages.ScalaPlayFrameworkServerCodegen"), date = "2026-02-09T20:57:39.376804970Z[Etc/UTC]", comments = "Generator version: 7.20.0-SNAPSHOT")
case class SubscriptionOption(
  id: Option[Long],
  active: Option[Boolean],
  valid: Option[Boolean],
  name: Option[String],
  price: Option[Double],
  quantity: Option[Int],
  threshold: Option[Int],
  visible: Option[Boolean],
  optionType: Option[SubscriptionOption.OptionType.Value],
  offerType: Option[SubscriptionOption.OfferType.Value]
)

object SubscriptionOption {
  implicit lazy val subscriptionOptionJsonFormat: Format[SubscriptionOption] = Json.format[SubscriptionOption]

  // noinspection TypeAnnotation
  object OptionType extends Enumeration {
    val MISSION = Value("MISSION")
    val LOCATION = Value("LOCATION")
    val OFFER = Value("OFFER")
    val NOTIFICATION = Value("NOTIFICATION")
    val REQUEST = Value("REQUEST")
    val STORAGES3 = Value("STORAGE_S3")
    val STORAGEDB = Value("STORAGE_DB")

    type OptionType = Value
    implicit lazy val OptionTypeJsonFormat: Format[Value] = Format(Reads.enumNameReads(this), Writes.enumNameWrites[this.type])
  }

  // noinspection TypeAnnotation
  object OfferType extends Enumeration {
    val VOUCHER = Value("VOUCHER")
    val COUPON = Value("COUPON")
    val PRODUCT = Value("PRODUCT")
    val MEDIA = Value("MEDIA")
    val EVENT = Value("EVENT")
    val DEVICE = Value("DEVICE")

    type OfferType = Value
    implicit lazy val OfferTypeJsonFormat: Format[Value] = Format(Reads.enumNameReads(this), Writes.enumNameWrites[this.type])
  }
}

File: server-aspnet-fastendpoints/src/Org.OpenAPITools/Models/SubscriptionOption.cs
namespace Org.OpenAPITools.Models;


/// <summary>
/// 
/// </summary>
public class SubscriptionOption 
{
    public long Id { get; set; }
    public bool Active { get; set; }
    public bool Valid { get; set; }
    public string Name { get; set; }
    public double Price { get; set; }
    public int Quantity { get; set; }
    public int Threshold { get; set; }
    public bool Visible { get; set; }
    
    /// <summary>
    /// Gets or Sets OptionType
    /// </summary>
    public enum OptionTypeEnum
    {
        
        /// <summary>
        /// Enum MISSIONEnum for MISSION
        /// </summary>
        MISSIONEnum = 1,
        
        /// <summary>
        /// Enum LOCATIONEnum for LOCATION
        /// </summary>
        LOCATIONEnum = 2,
        
        /// <summary>
        /// Enum OFFEREnum for OFFER
        /// </summary>
        OFFEREnum = 3,
        
        /// <summary>
        /// Enum NOTIFICATIONEnum for NOTIFICATION
        /// </summary>
        NOTIFICATIONEnum = 4,
        
        /// <summary>
        /// Enum REQUESTEnum for REQUEST
        /// </summary>
        REQUESTEnum = 5,
        
        /// <summary>
        /// Enum STORAGES3Enum for STORAGE_S3
        /// </summary>
        STORAGES3Enum = 6,
        
        /// <summary>
        /// Enum STORAGEDBEnum for STORAGE_DB
        /// </summary>
        STORAGEDBEnum = 7
    }

    public OptionTypeEnum OptionType { get; set; }
    
    /// <summary>
    /// Gets or Sets OfferType
    /// </summary>
    public enum OfferTypeEnum
    {
        
        /// <summary>
        /// Enum VOUCHEREnum for VOUCHER
        /// </summary>
        VOUCHEREnum = 1,
        
        /// <summary>
        /// Enum COUPONEnum for COUPON
        /// </summary>
        COUPONEnum = 2,
        
        /// <summary>
        /// Enum PRODUCTEnum for PRODUCT
        /// </summary>
        PRODUCTEnum = 3,
        
        /// <summary>
        /// Enum MEDIAEnum for MEDIA
        /// </summary>
        MEDIAEnum = 4,
        
        /// <summary>
        /// Enum EVENTEnum for EVENT
        /// </summary>
        EVENTEnum = 5,
        
        /// <summary>
        /// Enum DEVICEEnum for DEVICE
        /// </summary>
        DEVICEEnum = 6
    }

    public OfferTypeEnum OfferType { get; set; }
}


